src/shogun/classifier/svm/SVM.h

```diff
@@ -242,6 +242,21 @@ class CSVM : public CKernelMachine
 		void set_callback_function(CMKL* m, bool (*cb)
 				(CMKL* mkl, const float64_t* sumw, const float64_t suma));
 
+		/** @return whether mkl interleaving optimization is possible or not */
+		virtual bool supports_interleaving() const { return false; }
+
+		/** @return whether this svm is one-class or multi-class */
+		virtual bool is_one_class() const { return false; }
+
+		/** @return whether this svm is multiclass or one-class*/
+		bool is_multiclass() const { return !is_one_class(); }
+
+		/** @return whether this svm solves a regression problem or classification */
+		virtual bool is_regression() const { return false; }
+
+		/** @return whether this svm solves a classification problem or regression */
+		bool is_classification() const { return !is_regression(); }
```

Comment:
this is not used, so rather remove it

Reply:
it's used in MKLRegression

Reply:
I cannot find it in the diff

Reply:
I thought I would ask first. Currently `MKLClassification` doesn't check that the SVM is not a regression one.

Reply:
Actually, I just rememberd that we have an enum for this: `EProblemType` defined in `CMachine.h`  ... this can be reused here rather than adding some of the methods (interleaved needs to stay I guess)